pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/observables/tests/Test_Observables_TNCuda.cpp

```diff
@@ -0,0 +1,422 @@
+            }
+
+            SECTION("getWires") {
+                auto Z0 = std::make_shared<NamedObsT>(
+                    "PauliZ", std::vector<std::size_t>{0});
+                auto Z5 = std::make_shared<NamedObsT>(
+                    "PauliZ", std::vector<std::size_t>{5});
+                auto Z9 = std::make_shared<NamedObsT>(
+                    "PauliZ", std::vector<std::size_t>{9});
+
+                auto ham = HamiltonianT::create({0.8, 0.5, 0.7}, {Z0, Z5, Z9});
+
+                REQUIRE(ham->getWires() == std::vector<std::size_t>{0, 5, 9});
+
+                CHECK(ham->getCoeffs() ==
+                      std::vector<PrecisionT>{0.8, 0.5, 0.7});
+
+                CHECK(ham->getNumTensors() ==
+                      std::vector<std::size_t>{1, 1, 1});
+                CHECK(ham->getNumStateModes()[0] ==
+                      std::vector<std::size_t>{1});
+                CHECK(ham->getNumStateModes()[1] ==
+                      std::vector<std::size_t>{1});
+                CHECK(ham->getNumStateModes()[2] ==
+                      std::vector<std::size_t>{1});
+                CHECK(ham->getStateModes()[0][0] ==
+                      std::vector<std::size_t>{0});
+                CHECK(ham->getStateModes()[1][0] ==
+                      std::vector<std::size_t>{5});
+                CHECK(ham->getStateModes()[2][0] ==
+                      std::vector<std::size_t>{9});
+
+                CHECK(std::get<0>(ham->getMetaData()[0][0]) == "PauliZ");
+                CHECK(std::get<0>(ham->getMetaData()[1][0]) == "PauliZ");
+                CHECK(std::get<0>(ham->getMetaData()[2][0]) == "PauliZ");
+                CHECK(std::get<1>(ham->getMetaData()[0][0]) ==
+                      std::vector<PrecisionT>{});
+                CHECK(std::get<1>(ham->getMetaData()[1][0]) ==
+                      std::vector<PrecisionT>{});
+                CHECK(std::get<1>(ham->getMetaData()[2][0]) ==
+                      std::vector<PrecisionT>{});
+                CHECK(std::get<2>(ham->getMetaData()[0][0]) ==
+                      std::vector<ComplexT>{});
+                CHECK(std::get<2>(ham->getMetaData()[1][0]) ==
+                      std::vector<ComplexT>{});
+                CHECK(std::get<2>(ham->getMetaData()[2][0]) ==
+                      std::vector<ComplexT>{});
+            }
+
+            SECTION("Throw Errors") {
+                auto X0 = std::make_shared<NamedObsT>(
+                    "PauliX", std::vector<std::size_t>{0});
+                auto Z1 = std::make_shared<NamedObsT>(
+                    "PauliZ", std::vector<std::size_t>{1});
+
+                auto ob0 = HamiltonianT::create({TestType(0.5)}, {X0});
+
+                auto ob1 = HamiltonianT::create({TestType(0.5)}, {Z1});
+
+                REQUIRE_THROWS_AS(
+                    HamiltonianT::create({TestType(0.5), TestType(0.5)},
+                                         {ob0, ob1}),
+                    LightningException);
+            }
```

Comment:
Could you also add tests for HamiltonianObs of TensorProdObs?

Reply:
Tests added to line 356-400